onnxruntime/core/providers/cpu/rnn/deep_cpu_gru.cc

```diff
@@ -561,6 +561,8 @@ void UniDirectionalGru<T>::Compute(const gsl::span<const T>& inputs_arg,
   int32_t min_sequence_length = std::min(seq_length_, *std::min_element(sequence_lengths.cbegin(),
                                                                         sequence_lengths.cend()));
 
+  if (max_sequence_length == 0) return;
+
```

Comment:
This differs slightly in that we zero out the output for entries beyond the sequence length. So if max_sequence_length is 0 should we call fill_n to blank the output for consistency?

      for (int r = 0; r < batch_size_; r++) {
        if (step >= min_sequence_length && step >= sequence_lengths[r]) {
          if (output_sequence) {
            auto fill_output = output + r * hidden_size_;
            std::fill_n(fill_output, hidden_size_, T{});
          }

          continue;
        }

Reply:
>   for (int r = 0; r < batch_size_; r++) {
    if (step >= min_sequence_length && step >= sequence_lengths[r]) {
      if (output_sequence) {
        auto fill_output = output + r * hidden_size_;
        std::fill_n(fill_output, hidden_size_, T{});
      }

      continue;
    }

This logic zero out the output value between [sequence_lengths[r], max_sequence_length). As the max_sequence_length is already 0. Just return is consistent with the logic. 

However, it's a good practice to zero out the memory and make the output determined. I will zero out for the whole output.